packages/api-server/src/watch.ts

```diff
@@ -36,141 +33,44 @@ if (!process.env.REDWOOD_ENV_FILES_LOADED) {
   process.env.REDWOOD_ENV_FILES_LOADED = 'true'
 }
 
-const argv = yargs(hideBin(process.argv))
-  .option('debugPort', {
-    description: 'Port on which to expose API server debugger',
-    type: 'number',
-    alias: ['debug-port', 'dp'],
-  })
-  .option('port', {
-    description: 'The port to listen at',
-    type: 'number',
-    alias: 'p',
-  })
-  .parseSync()
+async function buildAndServe(options: BuildAndRestartOptions) {
+  const buildTs = Date.now()
+  console.log(chalk.dim.italic('Building...'))
+
+  if (options.clean) {
+    await cleanApiBuild()
+  }
 
-let httpServerProcess: ChildProcess
+  if (options.rebuild) {
+    await rebuildApi()
+  } else {
+    await buildApi()
+  }
 
-const killApiServer = () => {
-  httpServerProcess?.emit('exit')
-  httpServerProcess?.kill()
+  serverManager.restartApiServer()
```

Comment:
Are we okay not awaiting this async function?

I wonder if maybe that could contribute to #10876 but that's speculation and addressing that issue is not truly within the scope of this PR.

Reply:
I haven't tested this hypothesis but with large enough project, it's possible that this unwaited could try to kill and start api server at the same time. `await`ing makes sense as it prioritizes correctness over perf.

sidenote: I got distracted trying to setup [no-floating-promises](https://typescript-eslint.io/rules/no-floating-promises/) but configuration wasn't as straighforward and with all the hoops eslint process crashed with some OOM. I gave up there.